test/mocks/upstream/cluster_info.h

```diff
@@ -28,7 +28,7 @@ class MockClusterInfo : public ClusterInfo {
   MOCK_CONST_METHOD0(connectTimeout, std::chrono::milliseconds());
   MOCK_CONST_METHOD0(perConnectionBufferLimitBytes, uint32_t());
   MOCK_CONST_METHOD0(features, uint64_t());
-  MOCK_CONST_METHOD0(httpCodecOptions, uint64_t());
+  MOCK_CONST_METHOD0(http2Settings, const Http::Http2Settings&());
```

Comment:
by tests failing, do you mean unit test or integration tests?
it seems all the orignal unit tests run fine, but integration tests keep failing. i guess you already saw my earier comment on this.

not familiar with gtest/gmock, will try as you suggested and invest it.